tabulator/stream.py

```diff
@@ -444,8 +445,37 @@ def builtin_processor(extended_rows):
 
                 yield (row_number, headers, row)
 
+        def skip_negative_rows(extended_rows):
+            """
+            This processor will skip rows which counts from the end, e.g.
+            -1: skip last row, -2: skip pre-last row, etc.
+            Rows to skip are taken from  Stream.__skip_rows_by_numbers
+            """
+            rows_to_skip = [n for n in self.__skip_rows_by_numbers if n < 0]
+            if not rows_to_skip:
+                for row in extended_rows:
+                    yield row
+            else:
+                buffer_size = abs(min(rows_to_skip)) + 1
+                # collections.deque - takes O[1] time to push/pop values from any side.
+                buffer = deque()
+
+                # use buffer to save last rows
+                for row in extended_rows:
+                    buffer.append(row)
+                    if len(buffer) == buffer_size:
+                        yield buffer.popleft()
+
+                # Now squeeze out the buffer
+                last_row_number = buffer[len(buffer)-1][0]
+                # with last_row_number, we could transform negative row numbers to positive
+                rows_to_skip_positive = [last_row_number + 1 + n for n in rows_to_skip]
```

Comment:
a bit convoluted... 
why not something like:
```python
n = len(buffer)
for i, row in enumerate(buffer):
    if i-n not in rows_to_skip:
       yield row
```

Reply:
I used 'last_row_number' from `extended_row[0]` because 'extended_rows' were counted before any row deletions, so I avoid double deletions when two arguments are pointing on the same row (what you've been worried in the next comment)